### PR TITLE
feat: mandate Changes for the Saved card screen & SDK Button Loader changes

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Mandate Changes for the Saved card screen & SDK Button Loader changes

## How did you test it?

Via checking the customer_acceptance is going if we click on Checkbox on the saved cards screen.

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/Components/SaveDetailsCheckbox.res [hunk 1]
```diff
@@ -50,9 +51,13 @@ let make = (~isChecked, ~setIsChecked) => {
...
       <input type_={`checkbox`} onChange />
-      <div className={`checkmark CheckboxInput ${checkedState}`} />
-      <div className={`CheckboxLabel ${checkBoxLabelSate} ml-2`}>
-        {React.string(localeString.saveCardDetails)}
+      <div className={`checkmark CheckboxInput ${checkedState} mt-1`} />
+      <div className={`CheckboxLabel ${checkBoxLabelSate} ml-2 w-11/12`}>
+        {React.string(
+          showFields
+            ? localeString.saveCardDetails
+            : "By providing your card information, you allow to charge your card for future payments in accordance with their terms.",
```
Comment (akash-c-k) [inline src/Components/SaveDetailsCheckbox.res:59]: can we rephrase this? with their terms is not clear. Whose terms?

Comment (vsrivatsa-juspay) [inline src/Components/SaveDetailsCheckbox.res:59]:
```suggestion
            : localeString.cardTerms(options.business.name)
```

Comment (PritishBudhiraja) [reply src/Components/SaveDetailsCheckbox.res]: Updated

Comment (PritishBudhiraja) [reply src/Components/SaveDetailsCheckbox.res]: Updated

File: src/Components/SavedMethods.res [hunk 2]
```diff
@@ -153,9 +166,19 @@ let make = (
...
+    requiredFieldsBody,
+    empty,
+    complete,
+    customerMethod,
+    isCustomerAcceptanceRequired,
+  ))
   useSubmitPaymentData(submitCallback)
 
+  let conditionsForShowingSaveCardCheckbox =
+    !isGuestCustomer && list.payment_type !== SETUP_MANDATE && displaySavedPaymentMethodsCheckbox
```
Comment (ArushKapoorJuspay) [inline src/Components/SavedMethods.res:180]: Could you please use useMemo here since the list.payment_type might not get updated

**Review by ArushKapoorJuspay — APPROVED** [review]

**Review by prafulkoppalkar — APPROVED** [review]

**Pull request merged**